Blogging Prompts: Create prompts header view for create new action sheet

wargcm opened the pull request:
See #18402
Relies on PR #18453

## Description

Adds a blogging prompt header view to the create new action sheet

## Testing

To test:
- Enable blogging prompts feature flag
- Navigate to the 'My Site' tab
- Tap on the 'Create New' floating action button
- Observe prompts header view and verify:
  - Styling and that it matches the dashboard card
  - Text size
  - Dark mode
- Repeat verification with iPad

## Screenshots

### iPhone

| Before | After |
|--------|------|
| ![Simulator Screen Shot - iPhone 11 - 2022-04-26 at 16 59 41](https://user-images.githubusercontent.com/2454408/165395625-1736b246-db88-4b8d-a8d8-504d05c66700.png) | ![Simulator Screen Shot - iPhone 11 - 2022-04-26 at 17 07 16](https://user-images.githubusercontent.com/2454408/165395667-8f0fbfca-f838-433c-a798-9e8ed3f5d55a.png) |

### iPad
| Before | After |
|--------|------|
| ![Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2022-04-26 at 17 10 50](https://user-images.githubusercontent.com/2454408/165395782-c05f889f-d105-42c3-9e57-7bf33d398d99.png) | ![Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2022-04-26 at 17 11 46](https://user-images.githubusercontent.com/2454408/165395800-c73997c3-f658-4dc9-9955-e14856d725ec.png) |

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

wargcm committed:
Create prompts header view for bottom action sheet

See #18402

ScoutHarris reviewed (approved):
Just two minor requests please:
- The view for the divider/separator line in the xib - can you give that a name so it's clear what the view is for? All the other components have nice descriptive names. 
- Typically we set the height on lines such as this to `.hairlineBorderWidth` so it varies depending on device. We should probably do that here as on some devices the line is rather thick.

Otherwise looks good!

:shipit:

wargcm committed:
Update `BloggingPromptsHeaderView`'s divider view

Applies the following changes to the divider view:

- Add a name to the view
- Add a height constraint using `.hairlineBorderWidth`

wargcm commented:
Thanks for the feedback! I made the two suggested changes in: https://github.com/wordpress-mobile/WordPress-iOS/pull/18456/commits/698d0694ed3d5814b9498e541d5f3a7e27c37cf3
